app/ante/ante.go

```diff
@@ -91,17 +106,27 @@ func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
 	}, nil
 }
 
-func newCosmosAnteHandler(options HandlerOptions) sdk.AnteHandler {
+func newCosmosAnteHandler(options cosmosHandlerOptions) sdk.AnteHandler {
 	decorators := []sdk.AnteDecorator{}
 
 	decorators = append(decorators,
 		evmante.RejectMessagesDecorator{},   // reject MsgEthereumTxs
 		authante.NewSetUpContextDecorator(), // second decorator. SetUpContext must be called before other decorators
-		authante.NewRejectExtensionOptionsDecorator(),
 	)
+
+	if !options.isEIP712 {
+		decorators = append(decorators, authante.NewRejectExtensionOptionsDecorator())
+	}
```

Comment:
nit: looks like it's possible to add both a `Web3Tx` and `EthereumTx` option in a valid tx. I'd probably limit the number of extensions to just one to remove the chance for confusion. Maybe up on lines 72-73